esptool.py

```diff
@@ -294,6 +294,7 @@ def command(self, op=None, data=b"", chk=0, wait_response=True, timeout=DEFAULT_
         new_timeout = min(timeout, MAX_TIMEOUT)
         if new_timeout != saved_timeout:
             self._port.timeout = new_timeout
+            self._port.write_timeout = new_timeout
```

Comment:
Setting the same as the command timeout means that some commands (like erase) will have very long write timeouts (120 seconds).

I think it's probably safe to set write_timeout when the port is opened and not change it again. The most data esptool sends in a single command is 16KB (writing flash to ESP32). This can take a long time at a low baud rate, but looking at the pyserial code it looks like the write timeout is only until the OS buffers the write, and I think all OSes will buffer more than 16KB. So the write command itself should always return almost instantly, if the port is available and working correctly. Maybe the write timeout could be something like 3 seconds, to give some headroom?

Reply:
I think we can set a long timeout, 30 or 60 seconds for example, when opening the port. It will not affect normal use cases, only make sure esptool will finally exit in the rare scenario I mentioned in commit message.

Reply:
Because we have the "scan" feature if no port is given, it's useful to have esptool move on to a new port if it can't write to one for some reason. Maybe 10 seconds?

Reply:
Sure. I have changed to 10 seconds.